### PR TITLE
Swap Camera for Image Entity

### PR DESCRIPTION
**Pull request by prairiesnpr**
This swaps the Camera entity to an Image entity.

### CODE REVIEW

**Comment by Thomas55555** [comment]
- It seems like there is text missing for "Enable Image"

Screenshot 2023-07-11 204529

- Shouldn't we change the state of the image to `image_last_updated`. So that we have a timestamp there instead of, the mower's state?

File: custom_components/husqvarna_automower/entity.py [hunk 1]
```diff
@@ -3,11 +3,8 @@
 import logging
 from datetime import datetime
 
-from homeassistant.helpers.entity import DeviceInfo, Entity
-from homeassistant.helpers.update_coordinator import (
-    CoordinatorEntity,
-    DataUpdateCoordinator,
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/entity.py:9]: Ah good, that was unnecessary

**Comment by Thomas55555** [comment]
I've removed the 'HusqvarnaAutomowerStateMixin` class. Think, thats not relevant anymore. Is there a need to change the tests?

File: custom_components/husqvarna_automower/translations/en.json [hunk 2]
```diff
@@ -1,242 +1,243 @@
...
+        "state": {
+          "cleaning": "Mowing",
+          "docked": "Docked",
+          "error": "Error",
+          "idle": "Idle",
+          "off": "Off",
+          "on": "On",
+          "paused": "Paused",
+          "returning": "Returning to dock"
         }
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/translations/en.json:233]: Can be deleted ?!?

File: custom_components/husqvarna_automower/translations/de.json [hunk 3]
```diff
@@ -64,7 +64,7 @@
                 }
             }
         },
-        "camera": {
+        "image": {
             "mower": {
                 "state": {
                     "cleaning": "Mähen"
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/translations/de.json:70]: Can be deleted?!?!

File: custom_components/husqvarna_automower/translations/et.json [hunk 4]
```diff
@@ -1,243 +1,243 @@
...
+        "state": {
+          "cleaning": "Niidab",
+          "docked": "Dokitud",
+          "error": "Viga",
+          "idle": "[%key:common::state::idle%]",
+          "off": "[%key:common::state::off%]",
+          "on": "[%key:common::state::on%]",
+          "paused": "[%key:common::state::paused%]",
+          "returning": "Läheb laadimisjaama"
         }
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/translations/et.json:233]: Can be deleted?!?

File: custom_components/husqvarna_automower/translations/fr.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
                 }
             }
         },
-        "camera": {
+        "image": {
             "mower": {
                 "state": {
                     "cleaning": "Tonte"
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/translations/fr.json:30]: Can be deleted!?!?

File: custom_components/husqvarna_automower/translations/no.json [hunk 6]
```diff
@@ -49,7 +49,7 @@
                 }
             }
         },
-        "camera": {
+        "image": {
             "mower": {
                 "state": {
                     "cleaning": "Klipper"
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/translations/no.json:55]: Can be deleted!?!?!

File: custom_components/husqvarna_automower/translations/sv.json [hunk 7]
```diff
@@ -1,204 +1,204 @@
...
+        "state": {
+          "cleaning": "Klipper",
+          "docked": "Dockad",
+          "error": "Fel",
+          "idle": "Inaktiv",
+          "off": "Från",
+          "on": "Till",
+          "paused": "Pausad",
+          "returning": "Återvänder till dockan"
         }
```
Comment (Thomas55555) [inline custom_components/husqvarna_automower/translations/sv.json:194]: Can be deleted!??!

**Comment by Thomas55555** [comment]
Look good. It feels like the image entity is faster than the camera entity.....
I've added some comments to the translations. I think there can be deleted some stuff. Can you have a look at this?
Further more I removed the `HusqvarnaAutomowerStateMixin`
I've you're okay with that, I think we can merge this. I assume, you have tested migration from the old to the new version? I didn't do that.

**Comment by prairiesnpr** [comment]
> Look good. It feels like the image entity is faster than the camera entity..... I've added some comments to the translations. I think there can be deleted some stuff. Can you have a look at this? Further more I removed the `HusqvarnaAutomowerStateMixin` I've you're okay with that, I think we can merge this. I assume, you have tested migration from the old to the new version? I didn't do that.

I'll look this evening, I think the image entity is on average faster, but it seems to drag when using large images since we lose the ability to resize the image on the backend and that now has to be done in the front end.  I left the resize code in, hopefully that will get added back in HA and we can implement it.

I'm not certain about the state translations, thinking those should be removed in #511?  Will check to be sure.

I did test the migration, no issues found.

Didn't see any immediate concerns with [4dc8996](https://github.com/Thomas55555/husqvarna_automower/pull/512/commits/4dc8996372dd8cd9c497731a42f0b5c2cd1cbb57) , but want to load it up in a codespace and just double check.

**Comment by Thomas55555** [comment]
That are the state translation for the image. So they are not related to the other PR.

**Comment by prairiesnpr** [comment]
OK, this should be ready to merge now.

**Comment by fredoche1810** [comment]
Hello @prairiesnpr @Thomas55555 ,
Thans for your update.
Before, I could take a snapshot, save it as JPG, and add the JPG to a notification.
Now that the camera entity doesn't exist anymore, I can't take a snapshot.
How can I attach a dynamic image to my notification now ?
Thanks in advance,

**Comment by prairiesnpr** [comment]
> Hello @prairiesnpr @Thomas55555 , Thans for your update. Before, I could take a snapshot, save it as JPG, and add the JPG to a notification. Now that the camera entity doesn't exist anymore, I can't take a snapshot. How can I attach a dynamic image to my notification now ? Thanks in advance,

I'm using the same automation as before, just updated the camera entity to the image entity.

This is what I use now.

```
action:
  - variables:
      ha_url_var: !secret ha_url
      quest_update: '{{ states("sensor.aeinarr_error_quest_desc") }}'
  - service: notify.all_ios_devices
    data_template:
      message: '{{ state_attr("vacuum.aeinarr", "friendly_name") }} {{ quest_update }}'
      data:
        attachment:
          url: '{{ha_url_var}}{{ state_attr("image.aeinarr_map", "entity_picture") }}'
          content-type: jpg
          hide-thumbnail: false
  - service: notify.all_android_devices
    data_template:
      title: '{{ state_attr("vacuum.aeinarr", "friendly_name") }} Quest Update!'      message: '{{ state_attr("vacuum.aeinarr", "friendly_name") }} {{ quest_update }}'
      data:
        image: '{{ha_url_var}}{{ state_attr("image.aeinarr_map", "entity_picture") }}'
```

**Comment by fredoche1810** [comment]
Hello @prairiesnpr ,
Thanks for sharing your code. 
How do you manage !secret ha_url ?  
I tested your code with the internal url of homeassistant and it works if I'm on the LAN  
I tested with the external url and it works when I'm outside the LAN. 
How to make it working as well if connected to the LAN than when I'm outside the LAN ? 
Thanks,

**Comment by prairiesnpr** [comment]
In my setup, ha_url is just the external URL to my instance. I don't use the internal URL. That said, try DROPPING the URL, I don't think it's required.

**Pull request merged**